### PR TITLE
[#41] [Android] [UI] As a user, I can see the survey questions

### LINKED ISSUE #41: [Android] [UI] As a user, I can see the survey questions
## Why

When a user starts taking a survey, he/she will go through a set of questions of the selected survey. The mobile app should show how many questions there are for the survey so that the user can expect how long it is going to take and he/she can always stop at any point by dismissing the screen.

## Acceptance Criteria

- Display the background image
- Display the Dismiss button
    - Tapping the Dismiss button should navigate the user to the previous screen
- Display the Question Number label (Above the Question label)
- Display the Question label
- Display the Next button
- **DO NOT** display the Answer section yet.
    - There are many types of answers. There will be separate tickets to handle them.

## Design

[Take Survey](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=34%3A765&scaling=min-zoom)

### LINKED ISSUE #42: [Backend] As a user, I can see the survey questions
## Why

When a user starts taking a survey, he/she will go through a set of questions of the selected survey. The mobile app should show how many questions there are for the survey so that the user can expect how long it is going to take and he/she can always stop at any point by dismissing the screen.

## Acceptance Criteria

- Call `GET /api/v1/surveys/{{survey-id}}` endpoint to retrieve the survey details
- Map the response with the only necessary values for now

## Resources
[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0)

### PR DESCRIPTION
**Pull request by suho**
- Closes #41

## What happened 👀

Showed the survey questions screen with:
- Close button.
- Order label.
- Title.
- Next button.

## Insight 📝

- I also added the start survey button to open the survey questions screen.
- Added library for pager and constraint layout compose

## Proof Of Work 📹

Screenshot 2022-12-08 at 10 09 38

### CODE REVIEW

**Comment by hoangnguyen92dn** [comment]
@suho This PR suppose to close #41 instead? 🤔 

image

**Comment by suho** [comment]
@hoangnguyen92dn Updated! 🙈

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveydetail/SurveyDetailScreen.kt [hunk 1]
```diff
@@ -99,30 +98,60 @@ private fun SurveyDetailScreenContent(
         backgroundColor = Color.Transparent
     ) {
         Column(
+            verticalArrangement = Arrangement.SpaceBetween,
             modifier = Modifier
-                .padding(horizontal = 20.dp, vertical = 12.dp)
-                .verticalScroll(rememberScrollState())
+                .padding(horizontal = AppTheme.dimensions.mediumPadding, vertical = 12.dp)
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveydetail/SurveyDetailScreen.kt:103]: 👍

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                        brush = Brush.verticalGradient(
+                            colors = listOf(
+                                Color.Transparent,
+                                Color.Black.copy(alpha = 0.6f)
+                            )
+                        )
+                    )
+            )
+        }
+    }
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt:81]: I think the whole Image & Box widgets group could be extracted into a shared widget for 3 screens: Home Survey, Survey Detail & Survey Question screens. Could we do a code optimization here?

image

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt [hunk 3]
```diff
@@ -47,7 +48,15 @@ fun NavGraphBuilder.mainGraph(navController: NavController) {
...
                 navController.navigateUp()
+            },
+            onStartSurveyClick = {
+                navController.navigate(AppDestination.SurveyQuestions)
             }
         )
     }
+    composable(AppDestination.SurveyQuestions) {
+        SurveyQuestionsScreen(onCloseClick = {
+            navController.popBackStack()
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt:59]: Same, `navigateUp`

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt (hunk 2 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt]: Nice catch! Fixed in https://github.com/suho/kmm-ic/pull/114/commits/0d0b04b5795c2cf45387030653649ef22f39b1ef

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt (hunk 3 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt]: Fixed in https://github.com/suho/kmm-ic/pull/114/commits/0d0b04b5795c2cf45387030653649ef22f39b1ef

**Review by luongvo — APPROVED** [review]

**Pull request merged**